kafka-rest/src/test/java/io/confluent/kafkarest/response/StreamingResponseTest.java

```diff
@@ -294,69 +300,272 @@ public void testWriteToChunkedOutputAfterTimeout() throws IOException, Interrupt
         ResultOrError.error(
             ErrorResponse.create(
                 408,
-                "Streaming connection open for longer than allowed: Connection will be closed."));
+                "Streaming connection open for longer than allowed: "
+                    + "Connection will be closed."));
 
     ChunkedOutputFactory mockedChunkedOutputFactory = mock(ChunkedOutputFactory.class);
     ChunkedOutput<ResultOrError> mockedChunkedOutput = mock(ChunkedOutput.class);
 
     expect(clock.instant())
-        .andReturn(Instant.ofEpochMilli(0)); // stream start - input stream response (check)
-    expect(clock.instant())
-        .andReturn(Instant.ofEpochMilli(0)); // stream start - composing response (check)
-    expect(requestsMappingIterator.hasNext()).andReturn(true); // first message - OK
+        .andReturn(Instant.ofEpochMilli(0)); // stream start - input stream response
+
+    // first message
+    expect(clock.instant()).andReturn(Instant.ofEpochMilli(0));
+    expect(requestsMappingIterator.hasNext()).andReturn(true);
     expect(requestsMappingIterator.nextValue()).andReturn(request);
     expect(clock.instant())
         .andReturn(Instant.ofEpochMilli(1)); // first comparison duration.  within timeout
-
     expect(mockedChunkedOutputFactory.getChunkedOutput()).andReturn(mockedChunkedOutput);
-    mockedChunkedOutput.write(sucessResult);
     expect(mockedChunkedOutput.isClosed()).andReturn(false);
+    mockedChunkedOutput.write(sucessResult);
 
-    expect(requestsMappingIterator.hasNext()).andReturn(true); // is another message
-    expect(requestsMappingIterator.nextValue()).andReturn(request); // second message - bad gateway
+    // second message
+    expect(requestsMappingIterator.hasNext()).andReturn(true);
+    expect(requestsMappingIterator.nextValue()).andReturn(request);
     expect(clock.instant())
         .andReturn(Instant.ofEpochMilli(timeout + 5)); // second message beyond timeout
-
+    expect(mockedChunkedOutput.isClosed()).andReturn(false);
     mockedChunkedOutput.write(error);
+
+    // no third message
+    expect(requestsMappingIterator.hasNext()).andReturn(false);
+
+    requestsMappingIterator.close(); // call from thread executor
```

Comment:
Can't do an expect with something that returns a void, need to do the playback of the calls one by one :'( 

Add a comment explaining this, because this looks bonkers.